src/client/java/teammates/client/scripts/scalabilitytests/InstructorFeedbackResultsPageScalabilityTest.java

```diff
@@ -18,20 +25,150 @@
 
     private static final String DATA_FOLDER_PATH = "src/client/java/teammates/client/scripts/scalabilitytests/data/";
     private static final Logger log = Logger.getLogger();
+    private static DataBundle testDataMax;
 
     @Override
     protected void prepareTestData() {
         // the actual test data is refreshed before each test method
     }
 
     private void refreshTestData(int numStudents, int numQuestions) {
-        refreshTestData(
-                "InstructorFeedbackResultsPageScaleTest-" + numStudents + "Students" + numQuestions + "Questions.json");
+
+        // modify set of students for each test according to its requirements
+        if (testData.students.size() == 0) { // if current collection of students is empty
+            addStudents(numStudents);
+        } else if (testData.students.size() < numStudents) { // current number of students is below required
+            addStudents(numStudents - testData.students.size());
+        } else if (testData.students.size() > numStudents) { // current number of students is above required
+            decreaseNumOfStudents(numStudents);
+        }
+
+        // modify set of questions for each test according to its requirements
+        if (testData.feedbackQuestions.size() == 0) { // if current collection of questions is empty
+            addQuestions(numQuestions);
+        } else if (testData.feedbackQuestions.size() < numQuestions) { // current number of questions is below required
+            addQuestions(numQuestions - testData.feedbackQuestions.size());
+        } else if (testData.feedbackQuestions.size() > numQuestions) { // current number of questions is above required
+            decreaseNumOfQuestions(numQuestions);
+        }
+
+        // modify set of responses for each test according to its sets of students and questions
+        updateFeedbackResponses();
+
+        // perform test by reading each entity from datastore or persisting if absent
+        verifyOrPersistTestDataToDatastore();
+    }
+
+    // verify if entities for testing already exist in datastore
+    private void verifyOrPersistTestDataToDatastore() {
+        for (StudentAttributes student : testData.students.values()) {
+            try {
+                verifyPresentInDatastore(student);
```

Comment:
Even after clearing those issues, the test fails at https://github.com/TEAMMATES/teammates/pull/7289/files#diff-b40ea85664f4cc1c1c4378f8415c1a1fR191 because `CFResultsScT.instr` doesn't exist in the Datastore.